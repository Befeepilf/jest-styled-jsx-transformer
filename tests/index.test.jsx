import React from 'react';
import {render, screen} from '@testing-library/react';
import TestComponent from './TestComponent.jsx';
const {process} = require('../index.js');
const fs = require('fs');
const path = require('path');

describe("transformer", () => {
    const css = fs.readFileSync(path.resolve(__dirname + '/testStyles.css')).toString();
    const transformedCss = process(css, './testStyles.css');

    it("Transforms CSS", () => {
        expect(transformedCss).toMatchSnapshot();
    });

    describe("React component", () => {
        it("Elements have scoped className", () => {
            render(<TestComponent/>);

            const scopeId = transformedCss.match(/styles\.__hash = "(\d+)";/)[1];

            const h1 = screen.getByText('Headline');
            expect(h1.className).toMatch(new RegExp(`jsx-${scopeId}`));
            expect(h1.className).toMatch(/headline/);

            const p = screen.getByText('Lorem Ipsum');
            expect(p.className).toMatch(new RegExp(`jsx-${scopeId}`));
            expect(p.className).toMatch(/text/);
        });
    });
});