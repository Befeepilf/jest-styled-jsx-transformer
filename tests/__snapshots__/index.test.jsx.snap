// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformer Transforms CSS 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports[\\"default\\"] = void 0;
var styles = new String(\\".headline.jsx-723301647{color:red;}.text.jsx-723301647{text-align:center;}\\");
styles.__hash = \\"723301647\\";
var _default = styles;
exports[\\"default\\"] = _default;"
`;
